fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/insert/InsertIntoValuesAnalyzer.java

```diff
@@ -0,0 +1,148 @@
+
+import org.apache.doris.common.Pair;
+import org.apache.doris.nereids.CascadesContext;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.nereids.jobs.executor.AbstractBatchJobExecutor;
+import org.apache.doris.nereids.jobs.rewrite.RewriteJob;
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.rules.analysis.BindSink;
+import org.apache.doris.nereids.rules.expression.ExpressionRewrite;
+import org.apache.doris.nereids.rules.expression.ExpressionRewriteRule;
+import org.apache.doris.nereids.rules.expression.rules.ConvertAggStateCast;
+import org.apache.doris.nereids.rules.expression.rules.FoldConstantRuleOnFE;
+import org.apache.doris.nereids.rules.rewrite.MergeProjects;
+import org.apache.doris.nereids.rules.rewrite.OneRewriteRuleFactory;
+import org.apache.doris.nereids.rules.rewrite.PushProjectIntoOneRowRelation;
+import org.apache.doris.nereids.rules.rewrite.PushProjectIntoUnion;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.StatementScopeIdGenerator;
+import org.apache.doris.nereids.trees.plans.algebra.SetOperation.Qualifier;
+import org.apache.doris.nereids.trees.plans.logical.LogicalOneRowRelation;
+import org.apache.doris.nereids.trees.plans.logical.LogicalUnion;
+import org.apache.doris.nereids.types.DataType;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+
+/** InsertIntoValuesAnalyzer */
+public class InsertIntoValuesAnalyzer extends AbstractBatchJobExecutor {
+    public static final List<RewriteJob> INSERT_JOBS = jobs(
+            bottomUp(
+                    new InlineTableToUnionOrOneRowRelation(),
+                    new BindSink(),
+                    new MergeProjects(),
+                    // after bind olap table sink, the LogicalProject will be generated under LogicalOlapTableSink,
+                    // we should convert the agg state function in the project, and evaluate some env parameters
+                    // like encrypt key reference, for example: `values (aes_encrypt("abc",key test.my_key))`,
+                    // we should replace the `test.my_key` to real key
+                    new RewriteInsertIntoExpressions(ExpressionRewrite.bottomUp(
+                            ConvertAggStateCast.INSTANCE,
+                            FoldConstantRuleOnFE.PATTERN_MATCH_INSTANCE
+                    ))
+            )
+    );
+
+    public static final List<RewriteJob> BATCH_INSERT_JOBS = jobs(
+            bottomUp(
+                    new InlineTableToUnionOrOneRowRelation(),
+                    new BindSink(),
+                    new MergeProjects(),
+                    new PushProjectIntoUnion(),
+                    new PushProjectIntoOneRowRelation(),
```

Comment:
these rules is expensive? why inser jobs do not contain them?

Reply:
the BatchInsert need send StringLiteral to backend, and only support `alias(literal as xx)` or `alias(cast(literal as xx))`, but not support `alias(cast(slotRef as xx))` we should push down the cast into Union all OneRowRelation.